libc/string_catdup.c

```diff
@@ -0,0 +1,780 @@
+/*
+ * Phoenix-RTOS
+ *
+ * POSIX.1-2017 standard library functions tests
+ * HEADER:
+ *    - string.h
+ *
+ * TESTED:
+ *    - strcat()
+ *    - strncat()
+ *    - strdup()
+ *    - strmdup()
+ *
+ * Copyright 2023 Phoenix Systems
+ * Author: Mateusz Bloch
+ *
+ * This file is part of Phoenix-RTOS.
+ *
+ * %LICENSE%
+ */
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <errno.h>
+#include <ctype.h>
+#include <limits.h>
+#include <unity_fixture.h>
+#include "common.h"
+
+
+#define WORD_1          "Lorem "
+#define WORD_2          "ipsum"
+#define NON_ASCII_1     "\xe3\x83\x9e\xe3\x83\xaa\xe3\x82\xa2\xe3\x83\xbb"
+#define NON_ASCII_2     "\xe3\x82\xb9\xe3\x82\xaf\xe3\x82\xa6\xe3\x82\xa9\xe3\x83\x89\xe3\x83\x95\xe3\x82\xb9\xe3\x82\xab\xef\xbc\x9d\xe3\x82\xad\xe3\x83\xa5\xe3\x83\xaa\xe3\x83\xbc"
+#define SPECIALS_STRING "\n\t !\"#$%&'()*+,-./:;<=>?@[\\]^_`{|}~"
+#define SIZE_BUF        32
+#define SIZE_HUGE_BUF   150
+
+
+static char huge_buf[PATH_MAX];
+
+
+TEST_GROUP(string_cat);
+TEST_GROUP(string_dup);
+
+
+TEST_SETUP(string_cat)
+{
+	for (int i = 0; i < PATH_MAX - 1; i++) {
```

Comment:
I think placing it in test setup is unnecessary, many of cases don't use `huge_buf`. It could be done once somewhere. Above PATH_MAX could potentially be longer than `hugeTestStr` size, so yoo can check it and use modulo as next indexes.